storm-core/src/jvm/org/apache/storm/state/KeyValueState.java

```diff
@@ -52,4 +55,9 @@
      * @param key   the key
      */
     V delete(K key);
+
+    /**
+     * @return an iterator over all key value mappings.
+     */
+    Iterator<Map.Entry<K, V>> iterator();
```

Comment:
The KeyValueState can extend `Iterable<Entry<K, V>>` so that the state can be directly used in foreach constructs.